src/zkp/ecdsa_adaptor.rs

```diff
@@ -84,31 +88,116 @@ impl EcdsaAdaptorSignature {
     }
 }
 
+fn encrypt_helper<C: Signing>(
+    secp: &Secp256k1<C>,
+    msg: &Message,
+    sk: &SecretKey,
+    enckey: &PublicKey,
+    nonce_data: *mut ffi::types::c_void,
+) -> EcdsaAdaptorSignature {
+    unsafe {
+        encrypt_helper_with_nonce_fn(
+            secp,
+            msg,
+            sk,
+            enckey,
+            nonce_data,
+            ffi::secp256k1_nonce_function_ecdsa_adaptor,
+        )
+    }
+}
+
+fn encrypt_helper_with_nonce_fn<C: Signing>(
+    secp: &Secp256k1<C>,
+    msg: &Message,
+    sk: &SecretKey,
+    enckey: &PublicKey,
+    nonce_data: *mut ffi::types::c_void,
+    nonce_fn: ffi::EcdsaAdaptorNonceFn,
+) -> EcdsaAdaptorSignature {
+    let mut adaptor_sig = ffi::EcdsaAdaptorSignature::new();
+
+    unsafe {
+        debug_assert!(
+            ffi::secp256k1_ecdsa_adaptor_encrypt(
+                *secp.ctx(),
+                &mut adaptor_sig,
+                sk.as_c_ptr(),
+                enckey.as_c_ptr(),
+                msg.as_c_ptr(),
+                nonce_fn,
+                nonce_data,
+            ) == 1
+        );
+    };
+
+    EcdsaAdaptorSignature(adaptor_sig)
+}
+
 impl EcdsaAdaptorSignature {
-    /// Creates an adaptor signature along with a proof to verify the adaptor signature.
+    /// Creates an adaptor signature along with a proof to verify the adaptor signature,
+    /// using the ThreadRng random number generator to generate the auxiliary random data.
+    /// Requires compilation with "rand-std" feature.
+    #[cfg(any(test, feature = "rand-std"))]
     pub fn encrypt<C: Signing>(
         secp: &Secp256k1<C>,
         msg: &Message,
         sk: &SecretKey,
         enckey: &PublicKey,
     ) -> EcdsaAdaptorSignature {
-        let mut adaptor_sig = ffi::EcdsaAdaptorSignature::new();
+        let mut rng = thread_rng();
+        EcdsaAdaptorSignature::encrypt_with_rng(secp, msg, sk, enckey, &mut rng)
+    }
 
-        unsafe {
-            debug_assert!(
-                ffi::secp256k1_ecdsa_adaptor_encrypt(
-                    *secp.ctx(),
-                    &mut adaptor_sig,
-                    sk.as_c_ptr(),
-                    enckey.as_c_ptr(),
-                    msg.as_c_ptr(),
-                    None,
-                    ptr::null_mut(),
-                ) == 1
-            );
-        };
+    /// Creates an adaptor signature along with a proof to verify the adaptor signature,
+    /// using the provided random number generator to generate the auxiliary random data.
+    /// Requires compilation with "rand" feature.
+    #[cfg(any(test, feature = "rand"))]
+    pub fn encrypt_with_rng<C: Signing, R: Rng + CryptoRng>(
+        secp: &Secp256k1<C>,
+        msg: &Message,
+        sk: &SecretKey,
+        enckey: &PublicKey,
+        rng: &mut R,
+    ) -> EcdsaAdaptorSignature {
+        let mut aux = [0u8; 32];
+        rng.fill_bytes(&mut aux);
+        encrypt_helper(
```

Comment:
This can forward to `encrypt_with_aux_rand` instead.

Reply:
Good point fixed